workers/unity/Assets/Playground/Scripts/Cubes/CollisionProcessSystem.cs

```diff
@@ -0,0 +1,69 @@
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Playground
+{
+    [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
+    internal class CollisionProcessSystem : ComponentSystem
+    {
+        private struct LaunchableData
+        {
+            public readonly int Length;
+            public ComponentDataArray<CollisionComponent> Collision;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<CommandRequestSender<SpatialOSLauncher>> Sender;
+        }
+
+        [Inject] private LaunchableData launchableData;
+
+        protected override void OnUpdate()
+        {
+            for (var i = 0; i < launchableData.Length; i++)
+            {
+                Debug.Log("there are collisions");
+                var collision = launchableData.Collision[i];
+                var first = collision.own;
+                var second = collision.other;
+                var first_launcher = first.MostRecentLauncher;
+                var second_launcher = second.MostRecentLauncher;
+                if (first_launcher == 0 && second_launcher == 0)
+                {
```

Comment:
What's going on here?

Reply:
Handling different kinds of collisions, mostly determining whose score increases. But I agree that this warrants some comments.